lib/kernel/src/os.erl

```diff
@@ -313,7 +313,15 @@ mk_cmd(_,Cmd) ->
     %% We use an absolute path here because we do not want the path to be
     %% searched in case a stale NFS handle is somewhere in the path before
     %% the sh command.
-    {"/bin/sh -s unix:cmd", [out],
+    Shell = case filelib:is_regular("/bin/sh") of
```

Comment:
```suggestion
    Shell = case file:read_file_info("/bin/sh",[raw]) of
         {ok,#file_info{type=regular}} -> "/bin/sh";
```
Having thought about it a bit more, let us read the first regular like this instead so that we do not have to jump via the file server.

Reply:
Only for the first one in the regular case? Or for both to provide a consistent implementation?

Reply:
It is only really important for the first one, but may as well do it for both and add a comment about why we are doing it,

Reply:
> and add a comment about why we are doing it

Which I guess is ... a (much?) faster access ?

Reply:
Yes

Reply:
> Having thought about it a bit more, let us read the first regular like this instead so that we do not have to jump via the file server.
> ```Erlang
> -    Shell = case filelib:is_regular("/bin/sh") of
> +    Shell = case file:read_file_info("/bin/sh",[raw]) of
> +         {ok,#file_info{type=regular}} -> "/bin/sh";
> ```

I've reimplemented it this way and I've added a comment as suggested. 

As a side feedback, I was really confused at first when reading the doc of read_file_info. I was expecting to also have to support the case when `{ok,#file_info{type=symlink}}` is returned since /bin/sh is a symlink in many cases. However, my local testing has confirmed that it is not needed indeed.

The `file` documentation should be improved to describe this implementation detail as this is far from obvious and quite important in terms of semantics. I'm still not even sure to understand in which case `{ok,#file_info{type=symlink}}` can be returned when calling `read_file_info`.

Reply:
> I'm still not even sure to understand in which case {ok,#file_info{type=symlink}} can be returned when calling read_file_info.

I do not think it can. But the `file:read_link_info` documentation points to the same docs and when you use that you can get it. I'll see if I can make the docs clearer.